endpoint/endpoint.go

```diff
@@ -31,15 +31,18 @@ type Endpoint struct {
 	DNSName string
 	// The target the DNS record points to
 	Target string
+	// RecordType type of record, e.g. CNAME, A, TXT etc
+	RecordType string
 	// Labels stores labels defined for the Endpoint
 	Labels map[string]string
 }
 
 // NewEndpoint initialization method to be used to create an endpoint
-func NewEndpoint(dnsName, target string) *Endpoint {
+func NewEndpoint(dnsName, target, recordType string) *Endpoint {
```

Comment:
yes, were considering splitting into two different structs, one for endpoints (collected from k8s resources) and the other is the struct defined for dns provider records. However, afaik we should be able to define all dns record parameters (ttl, record type etc) in the resource annotations, so for now we will just stick to a single struct